src/cascadia/TerminalSettingsEditor/ColorSchemesPageViewModel.cpp

```diff
@@ -0,0 +1,158 @@
+#include "ColorSchemesPageViewModel.g.cpp"
+
+namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
+{
+    inline static constexpr std::array<std::wstring_view, 9> InBoxSchemes = {
+        L"Campbell",
+        L"Campbell Powershell",
+        L"Vintage",
+        L"One Half Dark",
+        L"One Half Light",
+        L"Solarized Dark",
+        L"Solarized Light",
+        L"Tango Dark",
+        L"Tango Light"
+    };
+
+    ColorSchemesPageViewModel::ColorSchemesPageViewModel(const Model::CascadiaSettings& settings) :
+        _settings{ settings },
+        _viewModelToSchemeMap{ winrt::single_threaded_map<Editor::ColorSchemeViewModel, Model::ColorScheme>() }
+    {
+        _MakeColorSchemeVMsHelper();
+        CurrentScheme(_AllColorSchemes.GetAt(0));
+    }
+
+    void ColorSchemesPageViewModel::UpdateSettings(const Model::CascadiaSettings& settings)
+    {
+        _settings = settings;
+
+        // We want to re-initialize our AllColorSchemes list, but we want to make sure
+        // we still have the same CurrentScheme as before (if that scheme still exists)
+
+        // Store the name of the current scheme
+        const auto currentSchemeName = CurrentScheme().Name();
+
+        // Re-initialize the color scheme list
+        _MakeColorSchemeVMsHelper();
+
+        // Re-select the previously selected scheme if it exists
+        const auto it = _AllColorSchemes.First();
+        while (it.HasCurrent())
+        {
+            auto scheme = *it;
+            if (scheme.Name() == currentSchemeName)
+            {
+                CurrentScheme(scheme);
+                break;
+            }
+            it.MoveNext();
+        }
+    }
+
+    void ColorSchemesPageViewModel::_MakeColorSchemeVMsHelper()
+    {
+        std::vector<Editor::ColorSchemeViewModel> allColorSchemes;
+
+        const auto colorSchemeMap = _settings.GlobalSettings().ColorSchemes();
+        for (const auto& pair : colorSchemeMap)
+        {
+            const auto scheme = pair.Value();
+            Editor::ColorSchemeViewModel viewModel{ scheme };
+            allColorSchemes.emplace_back(viewModel);
+
+            // We will need access to the settings model object later, but we don't
+            // want to expose it on the color scheme VM, so we store the reference to it
+            // in our internal map
+            _viewModelToSchemeMap.Insert(viewModel, scheme);
+        }
+
+        _AllColorSchemes = single_threaded_observable_vector<Editor::ColorSchemeViewModel>(std::move(allColorSchemes));
+    }
+
+    void ColorSchemesPageViewModel::RequestSetCurrentScheme(Editor::ColorSchemeViewModel scheme)
+    {
+        CurrentScheme(scheme);
+
+        // Update the state of the page
+        _PropertyChangedHandlers(*this, Windows::UI::Xaml::Data::PropertyChangedEventArgs{ L"CanDeleteCurrentScheme" });
+    }
+
+    void ColorSchemesPageViewModel::RequestEnterRename()
+    {
+        InRenameMode(true);
+    }

```

Comment:
This is a good example -- I had guessed that RequestEnterRename would fail if the scheme can't be edited... but here we are, with a 1:1 passthrough from request to action :)

Reply:
Same logic as earlier - to me this makes it clear that the page is the one requesting for a name change, and the view model is the one actually performing the change. Sure, we don't have logic right now that results in a rejection, but I'd still like it to be clear that the view model does have the 'authority' to do so (and in the future if we ever find the need for that logic, the name already makes sense)

Reply:
To expand on this, `RequestExitRename` has logic that might result in a rejection, but IMO it is not _because_ of this that the function is called `RequestXYZ`, it is called that way because the ViewModel is the one that ultimately decides what is going to happen - the page simply passes along that the user has requested for XYZ